### PR TITLE
add key data tags to keyboard

### PR DESCRIPTION
**Pull request by ghost**
each key needs its corresponding data

Testing notes:
- [x] keys properly wrap around after 12 tones

Screen Shot 2022-04-05 at 7 19 34 PM

### CODE REVIEW

File: keyboard.html [hunk 1]
```diff
@@ -36,25 +37,28 @@
     const allBlackKeysArray = Array.from(document.getElementsByClassName('black-key'))
     
     function addKeys(numKeys) {
-      const whiteKeysPerOctave = 6
+      const keysPerOctave = 11
```
Comment (dancingshell) [inline keyboard.html]:
```suggestion
      const keysPerOctave = keyNames.length
```

File: keyboard.html [hunk 2]
```diff
@@ -36,25 +37,28 @@
...
       let i = 0;
-      let whiteKeyCount = 0;
+      let keyCount = 0;
       while(i < numKeys) {
         // create a new rectangle div element
         const newDiv = document.createElement("div");
         newDiv.classList.add('white-key')
+        newDiv.setAttribute('key', `${keyNames[keyCount]}`)
```
Comment (dancingshell) [inline keyboard.html]: react uses `key` as a special attribute so you might want to name this `key-name` or `keyName` to avoid conflicts

Comment (dancingshell) [inline keyboard.html]: maybe making `addWhiteKey` its own function that increments `keyCount` and groups all the logic together will help clarify why its being incremented 2x

File: keyboard.html [hunk 3]
```diff
@@ -64,16 +68,17 @@
...
+      newDiv.setAttribute('key', `${keyNames[keyCount]}`)
       keyboard.appendChild(newDiv);
       sizeBlackKey(newDiv)
     }
 
-    function blackKeyCheck(whiteKeyCount) {
-      const whiteKeysNeedingBlackKeys = [0,1,3,4,5]
-      return whiteKeysNeedingBlackKeys.includes(whiteKeyCount)
+    function blackKeyCheck(keyCount) {
+      const whiteKeysNeedingBlackKeys = [0,1,2,5,7,9]
```
Comment (AllPurposeName) [inline keyboard.html]: This is telling me that the `keyNames` object ought to be expanded. off the top of my head I could imagine it as such:
```javascript
keys = {
 0: { "needsBlackKey": true, "name": "C" },
 ...
}
```

Comment (ghost) [reply keyboard.html]: i imagine i might need to do something along these lines down the road yeah...

Comment (ghost) [reply keyboard.html]: great callout thanks DJ!

**Pull request merged**